src/core/lib/security/credentials/tls/grpc_tls_certificate_provider.cc

```diff
@@ -452,3 +474,24 @@ void grpc_tls_certificate_provider_release(
   grpc_core::ExecCtx exec_ctx;
   if (provider != nullptr) provider->Unref();
 }
+
+grpc_status_code grpc_tls_certificate_key_match(
+    const char* private_key, const char* cert_chain, const char** error_details){
+  grpc_core::ExecCtx exec_ctx;
+  grpc_status_code code;
+  absl::StatusOr<bool> match_or = grpc_core::PrivateKeyAndCertificateMatch(private_key, cert_chain);
```

Comment:
It is a good practice to return early and avoid nesting if-else when possible.
This code can be updated to
```
if(!matched_or.status().ok()) {
  *error_details = gpr_strdup(...);
  return static_cast<grpc_status_code>(matched_or.status().code());
}
if (!*matched_or) {
  *error_details = "private key fails to match public key of leaf certificate";
  return GRPC_STATUS_INVALID_ARGUMENT;
}
*error_details = nullptr;
return GRPC_STATUS_OK;
```